toxcore/bin_pack.h

```diff
@@ -16,6 +16,23 @@ extern "C" {
 
 /**
  * @brief Binary serialisation object.
+ *
+ * Some notes on parameter order:
+ *
+ * - We pass the `obj` pointer as `this`-like pointer first to the callbacks.
```

Comment:
Isn't `Bin_Pack *bp` the `this` of `bin_[un]pack_()` functions, not `void *obj`?